### PR TITLE
support collecting feign params

### PR DESCRIPTION
**Pull request by zshit**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance

- Related issues
 #5233

### CODE REVIEW

File: docs/en/setup/service-agent/java-agent/README.md [hunk 1]
```diff
@@ -131,6 +131,8 @@ property key | Description | Default |
...
 `plugin.springmvc.collect_http_params`| This config item controls that whether the SpringMVC plugin should collect the parameters of the request, when your Spring application is based on Tomcat, consider only setting either `plugin.tomcat.collect_http_params` or `plugin.springmvc.collect_http_params`. Also, activate implicitly in the profiled trace. | `false` |
 `plugin.http.http_params_length_threshold`| When `COLLECT_HTTP_PARAMS` is enabled, how many characters to keep and send to the OAP backend, use negative values to keep and send the complete parameters, NB. this config item is added for the sake of performance.  | `1024` |
+`plugin.feign.trace_params`| This config item controls that whether the Feign plugin should collect the parameters of the request. | `false` |
+`plugin.feign.filter_length_limit`| When `TRACE_PARAMS` is enabled, how many characters to keep and send to the OAP backend, use negative values to keep and send the complete parameters, NB. this config item is added for the sake of performance.  | `1024` |
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/README.md:135]: What does NB represent?

File: apm-sniffer/apm-sdk-plugin/feign-default-http-9.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/feign/http/v9/DefaultHttpClientInterceptor.java [hunk 2]
```diff
@@ -81,6 +82,16 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         Tags.HTTP.METHOD.set(span, request.method());
         Tags.URL.set(span, request.url());
         SpanLayer.asHttp(span);
+        if (FeignPluginConfig.Plugin.Feign.TRACE_PARAMS) {
+            if (request.charset() != null && request.body() != null) {
+                String tagValue = new String(request.body(), request.charset());
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/feign-default-http-9.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/feign/http/v9/DefaultHttpClientInterceptor.java:87]: This is not just tracking the HTTP parameter, this is tracking the whole request BODY. The config name and document should be adjusted.
Also, you should do this based on HTTP request type, an HTTP request could include binary(such as a file), please use the format in the headers to filter the requests.

Comment (zshit) [reply apm-sniffer/apm-sdk-plugin/feign-default-http-9.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/feign/http/v9/DefaultHttpClientInterceptor.java:87]: thank you for your advice . I will try to solve it

File: docs/en/setup/service-agent/java-agent/README.md [hunk 3]
```diff
@@ -131,6 +131,9 @@ property key | Description | Default |
...
 `plugin.http.http_params_length_threshold`| When `COLLECT_HTTP_PARAMS` is enabled, how many characters to keep and send to the OAP backend, use negative values to keep and send the complete parameters, NB. this config item is added for the sake of performance.  | `1024` |
+`plugin.feign.collect_request_body`| This config item controls that whether the Feign plugin should collect the http body of the request. | `false` |
+`plugin.feign.filter_length_limit`| When `COLLECT_REQUEST_BODY` is enabled, how many characters to keep and send to the OAP backend, use negative values to keep and send the complete body.  | `1024` |
+`plugin.feign.supported_content_types_prefix`| When `COLLECT_REQUEST_BODY` is enabled and content-type start with SUPPORTED_CONTENT_TYPES_PREFIX, collect the body of the request , multiple paths should be separated by `,`  | `1024` |
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/README.md:136]: The default value is incorrect. Is Json the only default option?

File: test/plugin/scenarios/feign-scenario/src/main/java/org/apache/skywalking/apm/testcase/feign/controller/RestRequest.java [hunk 4]
```diff
@@ -43,6 +43,11 @@
     @Body("%7B\"id\": \"{id}\", \"userName\": \"{userName}\"%7D")
     User updateUser(@Param("id") int id, @Param("userName") String userName);
 
+    @RequestLine("PUT /modify/{id}")
+    @Headers("Content-Type: text/plain")
```
Comment (wu-sheng) [inline test/plugin/scenarios/feign-scenario/src/main/java/org/apache/skywalking/apm/testcase/feign/controller/RestRequest.java:47]: `text/plain` should be supported, right?

File: docs/en/setup/service-agent/java-agent/README.md (hunk 3 above)
Comment (zshit) [reply docs/en/setup/service-agent/java-agent/README.md:136]: default option 'application/json,text/ '  ?

Comment (wu-sheng) [reply docs/en/setup/service-agent/java-agent/README.md:136]: Personally, I prefer that. Any concern?

**Comment by wu-sheng** [comment]
Welcome to be 251st contributor.

**Pull request merged**